cssom-view-1/Overview.bs

```diff
@@ -1899,6 +1899,81 @@ When asked to <dfn export for=Document>run the resize steps</dfn> for a {{Docume
     <a attribute for=VisualViewport>width</a>, or <a attribute for=VisualViewport>height</a> properties changed since
     the last time these steps were run, <a>fire an event</a> named <a event>resize</a> at the {{VisualViewport}}.
 
+<h3 id=snap-events>Snap Events</h3>
+
+    This section integrates with the [[#scrolling-events|Scrolling events]].
+
+    Each {{Document}} associates each of its
+    <a spec=css-scroll-snap lt="scroll snap container">snap containers</a>
+    with at most one <dfn>snapchangingTargetBlock</dfn> and at most one
+    <dfn>snapchangingTargetInline</dfn> in the block and inline axes respectively.
+
+    Each {{Document}} associates each of its
+    <a spec=css-scroll-snap lt="scroll snap container">snap containers</a>
+    with at most one <dfn>snapchangedTargetBlock</dfn> and at most one
+    <dfn>snapchangedTargetInline</dfn> in the block and inline axes respectively.
+
+    When asked to <dfn export for=Document>update snapchanging targets</dfn>
+    for a <a spec=css-scroll-snap lt="scroll snap container">snap container</a>,
+    run these steps:
+
+    1. Let <var>doc</var> be the snap container's associated {{Document}}.
+    1. Let <var>scrollPosition</var> be the scroll offset resulting from applying
+        all pending scroll updates to the snap container.
+    1. Let <var>blockTarget</var> be the
+        <a>snapchangingTargetBlock</a> that <var>doc</var> associates with the snap
+        container.
+    1. Let <var>inlineTarget</var> be the <a>snapchangingTargetInline</a> that
+        <var>doc</var> associates with the snap container.
+    1. Let <var>newBlockTarget</var> be the element that the UA would snap the snap
+        container to from <var>scrollPosition</var> (or null if it would not snap to
+        any element) along the block axis.
+    1. Let <var>newInlineTarget</var> be the element that the UA would snap the snap
+        container to from <var>scrollPosition</var> (or null if it would not snap to
+        any element) along the inline axis.
+    1. If <var>newBlockTarget</var> is not the same element as
+        <var>blockTarget</var>:
+
+        1. Set <var>doc</var>'s <a>snapchangingTargetBlock</a> for the snap container
+            to <var>newBlockTarget</var>.
+        1. Append the snap container to <var>doc</var>'s
+            <a>pending snapchanging event targets</a>
+
+    1. If <var>newInlineTarget</var> is not the same element as
+        <var>inlineTarget</var>:
+        1. Set <var>doc</var>'s <a>snapchangingTargetInline</a> for the snap container
+            to <var>newInlineTarget</var>.
+        1. Append the snap container to <var>doc</var>'s
+            <a>pending snapchanging event targets</a>
```

Comment:
So if both inline and block change, you append it twice, which means that you fire two snapchanging events, that seems wrong?

Reply:
Yes I've corrected this to say that it should be appended only if it isn't already in the list.